src/diff.jl

```diff
@@ -37,3 +41,19 @@ differentiate(p::RationalPoly, x::PolyVar) = (differentiate(p.num, x) * p.den -
 function differentiate{C}(p::PolyType{C}, xs::Vector{PolyVar{C}})
     [differentiate(p, x) for x in xs]
 end
+
+# In Julia v0.5, it returns Any
+Base.promote_op{C}(differentiate, ::Type{PolyVar{C}}, ::Type{PolyVar{C}}) = Term{true, Int}
```

Comment:
I don't know the details of `promote_op` but the first argument should be a function and it does inference to check what is the output type of this function with a given input type. Here `differentiate` is a variable that holds the function object.